File: game/src/main/java/me/firecloud/game/rule/GameContext.java
/**
 * 
 */
package me.firecloud.game.rule;

import java.util.List;

/**
 * @date 2012-10-12
 * 
 */
public interface GameContext {
	public Object get(String epxression);

	public List<Object> getList(String expression);

	public GameContext set(String expression, Object value);
}

File: card/src/main/scala/me/firecloud/card/rule/GameContext.scala
/**
 *
 */
package me.firecloud.card.rule
import scala.collection.immutable.List
import java.util.Map
import java.util.HashMap

/**
 * @author kkppccdd
 * @email kkppccdd@gmail.com
 * @date Oct 19, 2012
 *
 */
class GameContext {
  val players: List[Player]
  val dealer: Dealer

  val circles: Map[String, ActorCircle] = new HashMap[String, ActorCircle]

  def getCircle(name: String): ActorCircle = {
    return circles.get(name)
  }
}